lib/vnet/daemon/client_darwin.h

```diff
@@ -34,4 +37,39 @@ int DaemonStatus(const char *bundle_path);
 // the login item for the daemon.
 void OpenSystemSettingsLoginItems(void);
 
+typedef struct StartVnetRequest {
+  const char *bundle_path;
+  VnetConfig *vnet_config;
+} StartVnetRequest;
+
+typedef struct StartVnetResult {
+  bool ok;
+  const char *error_domain;
+  const char *error_description;
+} StartVnetResult;
+
+// StartVnet spawns the daemon process. Only the first call does that,
+// subsequent calls are noops. The daemon process exits after the socket file
+// in request.vnet_config.socket_path is removed. After that it can be spawned
+// again by calling StartVnet.
+//
+// Blocks until the daemon receives the message or until the client gets
+// invalidated.
+//
+// After calling StartVnet, the caller is expected to call InvalidateDaemonClient
+// when a surrounding context in Go gets canceled, to check outResult.ok to see
+// if the client was able to connect to the daemon, and to free strings in StartVnetResult.
+void StartVnet(StartVnetRequest *request, StartVnetResult *outResult);
+
+// InvalidateDaemonClient closes the connection to the daemon and unblocks
+// any calls awaiting a reply from the daemon.
+void InvalidateDaemonClient(void);
+
+@interface VNEDaemonClient : NSObject
```

Comment:
FYI: C and C++ I can do an alright job reviewing (I hope?), but "proper" ObjC is a mystery to me.